packages/txservice/src/dispatch.ts

```diff
@@ -455,14 +479,35 @@ export class TransactionDispatch extends ChainRpcProvider {
 
     // Increment transaction # attempts made.
     transaction.attempt++;
-    // Set the timestamp according to when we last submitted.
-    transaction.timestamp = Date.now();
 
     // Send the tx.
-    const result = await this.sendTransaction(transaction);
-    // If we end up with an error, it should be thrown here.
-    if (result.isErr()) {
-      const error = result.error;
+    try {
+      const response = await this.sendTransaction(transaction);
+      // Add this response to our local response history.
+      if (transaction.hashes.includes(response.hash)) {
+        // Duplicate response? This should never happen.
+        throw new TransactionProcessingError(TransactionProcessingError.reasons.DuplicateHash, method, {
+          chainId: this.chainId,
+          response,
+          transaction: transaction.loggable,
+        });
```

Comment:
this would be such a weird error so if there's *anything* else that should be logged (request context/method context maybe) along with the error it should be included in the error context

Reply:
the response is everything here
also `transaction.loggable`
all we would need is hash in this case.

i've never seen this happen, it's just to enforce that transaction.hashes is a set of unique hashes... i imagine you could only break this by unit testing improperly